### PR TITLE
Move 'Nix flake submodule sync' to a separate workflow

### LINKED ISSUE #2718: [nix] Keep haskell-backend and llvm-backend flake inputs in sync with submodules
We need to add the following to the script which updates the haskell/llvm submodules and pushes to master:

```
nix flake lock \
  --override-input haskell-backend github:runtimeverification/haskell-backend/<hash> \
  --override-input llvm-backend github:runtimeverification/llvm-backend/<hash> \
  --recreate-lock-file
```

@F-WRunTime not sure where exactly we update and push the submodules?

### PR DESCRIPTION
**Pull request by goodlyrottenapple**
fixes #2718

### CODE REVIEW

**Review by F-WRunTime — APPROVED** [review]

**Comment by goodlyrottenapple** [comment]
@ehildenb Can you approve? Also, It seems when I open a new branch/PR against `develop` directly, none of the CI runs get triggered.

**Comment by dwightguth** [comment]
Will this fix the issue where submodule bump PRs fail tests until flake.lock is updated?

**Comment by goodlyrottenapple** [comment]
yes

**Review by dwightguth — APPROVED** [review]

**Comment by dwightguth** [comment]
This just needs to have the expected checks adjusted before it can be merged.

**Comment by dwightguth** [comment]
@F-WRunTime I will let you make that change.

File: .github/workflows/udapte-deps.yml
Comment (ehildenb) [inline .github/workflows/udapte-deps.yml]: This block should never fire now, and can be removed.

Also, we are using `ref=${{ github.ref }}` in this bash block, when we could probably just use `${GITHUB_HEAD_REF}` variable directly with is set by the github workflow?

Either way, this `if` statement doeeesn't make sense any more in a branch push, but it can be removed in follow-up.

File: .github/workflows/udapte-deps.yml
Comment (ehildenb) [inline .github/workflows/udapte-deps.yml]: This is also basically fully unnecessary now that I read it. I'll make a separate PR cleaning up some of this stuff and the maven steps probably.

**Review by ehildenb — APPROVED** [review]

**Comment by ehildenb** [comment]
I've made the adjustments to the required status checks.

File: .github/workflows/udapte-deps.yml
Comment (ehildenb) [reply .github/workflows/udapte-deps.yml]: I'll make these changes in my local branch and make a PR with updates.

File: .github/workflows/udapte-deps.yml
Comment (ehildenb) [reply .github/workflows/udapte-deps.yml]: Same, I'll make these changes in my local branch then open a PR with updates.

**Pull request merged**
